lib/integration/fsc.ts

```diff
@@ -1,15 +1,63 @@
 import { TIME_ZONE } from "config/config";
 import { calculateMondayOffset } from "lib/integration/common";
 import { DateTime } from "luxon";
-import { FscWeekSchedule, FscWeekScheduleResponse } from "types/integration/fsc";
+import {
+    DetailedFscWeekSchedule,
+    FscActivityDetail,
+    FscActivityDetailResponse,
+    FscWeekSchedule,
+    FscWeekScheduleResponse,
+} from "types/integration/fsc";
 
 function fscWeekScheduleUrl(fromDate: DateTime) {
     return `https://fsc.no/api/v1/businessunits/8/groupactivities?period_start=${fromDate.toUTC()}&period_end=${fromDate
         .plus({ week: 1 })
         .toUTC()}`;
 }
 
-export async function fetchFscWeekSchedule(weekOffset: number): Promise<FscWeekSchedule> {
+function fscActivityDetailUrl(activityId: number) {
+    return `https://fsc.no/api/v1/products/groupactivities/${activityId}`;
+}
+
+async function fetchDetailedFscWeekSchedule(fscWeekSchedule: FscWeekSchedule): Promise<DetailedFscWeekSchedule> {
+    const activityIds = Array.from(new Set(fscWeekSchedule.map((fscClass) => fscClass.groupActivityProduct.id)));
+    const activityDetails: Map<number, FscActivityDetail> = new Map();
+    await Promise.all(
```

Comment:
What about mapping the whole schedule and for each class check if already fetched, if not then fetch and store in map, and finally return the details. That would cut some loops. Performance gain is of course minimal, so it's more a stylistic preference.
(This approach requires that the promises are stored and then awaited only after all classes have been checked, so that we avoid fetching duplicates)

Reply:
I made somthing fancy here as we discussed. Should be pretty solid